Cookbook.Application/Validators/RecipeValidator.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Cookbook.Application.Database;
+using Cookbook.Domain.Models;
+using FluentValidation;
+
+namespace Cookbook.Application.Validators //is this the right place for validators?
+{
+    public class RecipeValidator : AbstractValidator<Recipe>
+    {
+        private readonly IRecipebookRepository recipebookRepository;
+        public RecipeValidator(IRecipebookRepository recipebookRepository)
+        {
+            this.recipebookRepository = recipebookRepository;
+            
+            RuleFor(x => x.Title).NotEmpty();
```

Comment:
Ami kicsit erdekes, es nem tudom hogy ezzel mit lehet kezdeni, de elobb hal el egy exception-on minthogy ide eljutna. Kiprobaltam hogy egy request-bol kivettem a title-t, es elso lepesben megprobalja deserializalni a CreateRecipeRequest-et, es az nem sikerul neki es dob egy hibat. Ha az sikerul neki utana hajtja ezt csak vegre (pl a greaterThan(0)-t). Szoval hiaba van itt NotEmpty minden property-re, nem innen jon a hiba, hanem a deserializalasbol.

Reply:
Ja kozben rajottem hogy azert mert a CreateRecipeRequest-et (a bejovo JSON-t) egyaltalan nem validalod, csak a Recipe-t ami megy az adatbazisba. 

Valami input validation kellene szerintem. Nem tudom pontosan hogy hogy szednem szet, de a CreateRecipeRequest-ben kene validalni hogy a NumberOfPortions az tenyleg int, a Title property tenyleg letezik, az Ingredients tenyleg egy array, stb. 
Most mindenfele elbaszott json-t bekuldhetek es vagy internal exception-t kapok (int.Parse elhal) vagy mar a deserialization is elhal. Egyik sem nice response. 
Aztan a mostani validation maradhat, ez inkabb ilyen business logic validation, hogy a number of portion nem zero, meg nem ures a title, a masik meg lehetne ilyen basic input validation hogy a tipusok stimmelnek meg a property-k megvannak.

Reply:
![image](https://github.com/Andrew00000/Cookbook/assets/62332857/bf24f7f6-956a-43b0-b251-a97030f8b52c)
en ezt kapom vissza ha kihagyom a titlet vagy ha mondjuk intet rakok bele. vagy ez nem jo? itt nem biztos h ertem mi a baj

Reply:
beszeltuk de ezt egyelore elengedtem kene hozza custombinding majd ha kesz visszanezek ra